### DISCUSSION TITLE
Blazorise licensing changes

### DISCUSSION THREAD

**Discussion by fileman** [abpframework/abp#17248] [General] [answered]
Hi,
maybe I have lost something related to this https://blazorise.com/news/2023-changes-to-blazorise-plans-and-licensing
We cant use blazorise for free if not "individual users" ?

**Comment by hikalkan** [comment]
I read the post. And yes, if you are a company, you must buy a license. From the post:

>...we're implementing new licensing terms starting July 15th, 2023:
>
>- The Community license, which will remain free, will be available exclusively for individual users.
>- All companies must purchase a Commercial license regardless of their size or revenue.

**Comment by ebicoglu** [comment, marked as answer]
we have created an internal issue to resolve this in the upcoming versions

**Reply by fileman** [reply]
https://blazor.radzen.com/ can be an alternative?

**Reply by hikalkan** [reply]
Radzen has a huge potential to become a paid product, IMHO.

**Reply by fileman** [reply]
so its time for ABP to have builtin controls like other frameworks... OMG

**Reply by hikalkan** [reply]
Unfortunately, it is out of this project's scope, at least for now. We can't do everything :)

**Reply by fileman** [reply]
Is this still valid https://github.com/abpframework/abp/discussions/14486#discussioncomment-4018147 ?

**Reply by gfo2007** [reply]
@hikalkan radzen have their own rapid application tool they've developed with paid subscription. So I doubt they will change the license on their blazor components. The confusion with blazorise right now is that, it is no more free for commercial use. But abp framework which uses blazorise is free for commercial use.

**Reply by genoher** [reply]
https://www.fluentui-blazor.net/

**Comment by gfo2007** [comment]
@fileman I never knew this. Thanks for the info. that makes it very complicated. Abp framework which free for commercial has blazorise which is not free for commercial use. I wish the abp team used **Mudblazor** which is free for commercial use and has the most stars on github as a blazor component library.

There is an article in the abp community for replacing blazorise with Mudblazor but anytime I try it, I get a blank screen.

**Reply by fileman** [reply]
If I'm not mistaken you can integrate with other libraries, but not replace Blazorise

**Reply by gfo2007** [reply]
@fileman have you tried using any other library? How did it work out?

**Reply by fileman** [reply]
I am using syncfusion, you can mix controls in components but not in forms.

**Reply by Fronix** [reply]
Yeah we still can't use ABP because of this small annoying detail, we already use Mudblazor but we can't remove Blazorise so we can't use ABP...

**Reply by gfo2007** [reply]
@Fronix  I hope the Volosoft team addresses this issue. You cannot have an open source project free for commercial use yet integrate it with a library that is not free for commercial use. It creates a lot of issues for the users.

**Comment by Tony20221** [comment]
Remove Blazorise and use MudBlazor instead. It makes no sense that a free open source project is reliant on a paid only (for commercial use) third party library or UI component when the open source project is used for commercial use.

**Comment by BenWhite27** [comment]
My vote is for MudBlazor too. I already completely replace the lepton theme with a custom one that removes as much blazorise as I can.  Have used MudBlazor exclusively for many years now, I found it the easiest component library to work with.

**Reply by OnlyOneBDS** [reply]
How do you remove Blazorise? Like what is the process of doing it?

**Reply by BenWhite27** [reply]
I started with this community post series https://community.abp.io/posts/mudblazor-theme-in-abp-blazor-webassembly-ae23zz17.  What I've done has moved away from that a little bit now.  I moved a bunch of code directly into my blazor project and haven't implemented some of the ABP features yet like Settings or Feature Management.

I've also made some modifications to the MudCrudPageBase to better align with my own developer preferences like not having MudDialogs defined inline on the CRUD page.  Allowing me to reuse _create_ and _update_ dialogs across the app from different screens.

The community posts are a bit old now but it's not that bad to bring them up to date with the help of the migration guides.

It'd be nice to fully remove the blasorise reference in the code but if I remember right it's a baked in reference in some of the ABP modules like settings and feature management that are in the reference chain.  
One day I'll take another look at that, but with my current implementation I am able to remove this from my module file.
```
context.Services
            .AddBootstrap5Providers()
            .AddFontAwesomeIcons(); 
```

I do wish there was a startup template where you can get the good stuff like cross cutting concerns, Unit of Work, Repositories, Event Management, Background Jobs, etc. without the opinionated UI.

On a side note, I also use the HttpApi host to run the Blazor front end and recently had success with web assembly pre-rendering with MudBlazor components.  Hopefully some of the new Blazor United stuff gets the spotlight in ABP 8 moving forward.

**Reply by OnlyOneBDS** [reply]
How did you move the MudDialogs away from the CRUD pages? I'm having a hard time trying to implement that..

**Comment by dongfo** [comment]
you can use vue , many free ui such as antdesign,element,vuetify  .

**Reply by fileman** [reply]
this issue is for Blazor UI native controls

**Comment by AndersChen123** [comment]
Maybe [Fluent UI Blazor](https://www.fluentui-blazor.net/) is a good choice.

**Comment by wujunchengman** [comment]
Maybe, can use [Masa.Bazor](https://docs.masastack.com/blazor/getting-started/installation)

**Comment by sgf** [comment]
I agree that Mudblazor is an amazing library, but the quality of its implementation is not as good as blazorise, and blazorise has more features, such as RichText integration,MarkDown Edit, BulmaCSS, etc.

There is no substitute for blazorise in terms of the features it implements. It is the best.
